### PR TITLE
feat(catalog): introduce clean_watermark_index_in_pk for state cleaning

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://raw.githubusercontent.com/risingwavelabs/risingwave/17af8a747593ebdbfa826691daf75bdab7d14fa0/.github/contributor-license-agreement.txt).

## What's changed and what's your intention?

as title 

part of https://github.com/risingwavelabs/risingwave/issues/18802

## Checklist

- [ ] I have written necessary rustdoc comments.
- [ ]  I have added necessary unit tests and integration tests.
- [ ]  I have added test labels as necessary. 
- [ ]  I have added fuzzing tests or opened an issue to track them. 
- [ ]  My PR contains breaking changes. 
- [ ]  My PR changes performance-critical code, so I will run (micro) benchmarks and present the results. 
- [ ]  My PR contains critical fixes that are necessary to be merged into the latest release. 

## Documentation

- [ ]  My PR needs documentation updates. 

<details>
<summary><b>Release note</b></summary>

</details>

### CODE REVIEW

File: src/frontend/src/catalog/table_catalog.rs [hunk 1]
```diff
@@ -700,6 +703,7 @@ impl From<PbTable> for TableCatalog {
             webhook_info: tb.webhook_info,
             job_id: tb.job_id.map(TableId::from),
             engine,
+            clean_watermark_index_in_pk: None, // TODO: fill this field
```
Comment (st1page) [inline src/frontend/src/catalog/table_catalog.rs:706]:
```suggestion
            clean_watermark_index_in_pk: tb.clean_watermark_index_in_pk.map(|x| x as usize)
```

File: proto/catalog.proto [hunk 2]
```diff
@@ -482,6 +482,8 @@ message Table {
   // Table Engine, currently only support hummock and iceberg
   optional Engine engine = 43;
 
+  optional int32 clean_watermark_index_in_pk = 44;
```
Comment (st1page) [inline proto/catalog.proto:485]: please add some comment that notice developer when it is "None", the value should be the first field in the pk

Comment (Li0k) [reply proto/catalog.proto:487]: thanks, Done

**Review by chenzl25 — COMMENTED** [review]
Please add a sql backend migration file. See https://github.com/risingwavelabs/risingwave/blob/6b10d92ecbacec65f361823799ef6470630ee316/src/meta/model/migration/README.md#L23

**Review by chenzl25 — APPROVED** [review]
LGTM

**Pull request merged**